src/core/geometry/qgsgeometryeditutils.cpp

```diff
@@ -105,6 +107,20 @@ Qgis::GeometryOperationResult QgsGeometryEditUtils::addPart( QgsAbstractGeometry
     return Qgis::GeometryOperationResult::InvalidInputGeometryType;
   }
 
+  // TriangulatedSurface does not inherit from QgsGeometryCollection
+  if ( QgsWkbTypes::flatType( geom->wkbType() ) == Qgis::WkbType::TIN )
+  {
+    QgsTriangulatedSurface *triangulatedSurface = qgsgeometry_cast<QgsTriangulatedSurface *>( geom );
+    if ( triangulatedSurface )
+    {
+      QgsTriangle *triangle = qgsgeometry_cast<QgsTriangle *>( part.get() );
+      if ( triangle )
+      {
+        triangulatedSurface->addPatch( triangle );
+      }
+    }
```

Comment:
missing return Qgis::GeometryOperationResult::Success in the success code path and Qgis::GeometryOperationResult::InvalidInputGeometryType if the part is not a triangle ?

Reply:
Indeed. Fixed.